private/src/gst/gstavbvideosrc.cpp

```diff
@@ -108,6 +116,10 @@ static void gst_avbvideosrc_set_property(GObject *object, guint prop_id,
     if (prop_id == PROP_STREAM_NAME) {
         free(avbvideosrc->stream_name);
         avbvideosrc->stream_name = g_value_dup_string(value);
+
+        if (strcasestr(avbvideosrc->stream_name, "mpegts") == NULL)
```

Comment:
But one can create a string like "video.from.cam.1", and not state that is mpegts or h264, can't they?

Reply:
Hmm, yea, I was only looking at the existing video profiles that either have mpegts or not. So yes, if someone creates a new video profile then this would break. 
Also, it will be the user's responsibility to pass the stream type _corresponding_ to the stream name (from the video profile), otherwise it will just break.